test/zero/unit/pubsub_test.clj

```diff
@@ -76,3 +78,30 @@
     (finally
       (pubsub/unsubscribe project subscription)
       (pubsub/delete-topic project topic))))
+
+(def notification-bucket
+  "A GCS bucket that publishes to a PubSub topic on file upload events."
+  "test-storage-notifications")
+
+(def notification-subscription
+  "The subscription to the PubSub topic that the notification-bucket publishes to."
+  "test-storage-notification-subscription")
+
+(deftest bucket-notification-test
```

Comment:
Looks good. Could you randomise the upload location to avoid sporadics from concurrent testing? There's a macro`zero.tools.fixtures/with-temporary-gcs-folder` that does this for you:
- It gives you a gcs object prefix to prepend to your object names
- It cleans up all resources with that prefix on return

See `test/zero/integration/v1_endpoint_test.clj:61` for an example

Reply:
I see that macro creates a temp folder in the "broad-gotc-dev-zero-test" bucket, but I used a different bucket for this test since it needs to be configured to publish messages to a topic. I can try making the bucket a parameter of the macro instead...

Reply:
Ah OK, if it's too much work don't worry. It's more a case of avoiding weird errors if one of us happens to run this test at the same time (or concurrently in CI)

Reply:
That makes sense! I attempted to refactor the existing macro but didn't get very far so I updated the test without it 😅